cmd/soroban-cli/src/tx/builder/transaction.rs

```diff
@@ -0,0 +1,66 @@
+use soroban_env_host::xdr::{self, Memo, SequenceNumber, TransactionExt};
+
+use super::{Error, MuxedAccount, Operation};
+
+#[derive(Debug, Clone)]
+pub struct Transaction {
```

Comment:
We made a conscious decision, lost to tribal knowledge, with the Rust stellar-xdr lib to attempt to go as far as possible with using XDR types, and filling gaps in their interface as we go, so that we didn't end up with a similar burden in the future that we have discovered on other SDKs where these layers end up diverging from the networks definition.

It's part of the reason the XDR types support conversion to and from things like strkeys (in addition to supporting the JSON encoding).

It looks like in this PR we're introducing a layer that looks much like other Stellar SDKs, similar to but different to the underlying XDR layer, that'll need to be maintained. Maybe that's no bigger than if the layer didn't exist, but once https://github.com/stellar/rs-stellar-xdr/pull/396 is adopted I think we could chisel this change back and remove the need for most conversions, and maybe the layer should get a lot thinner.

Reply:
I removed it in favor an extension trait.